.github/workflows/maven-build.yml

```
@@ -1,23 +1,25 @@
-# This workflow will build a Java project with Maven
-# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
-
 name: Java CI Build with Maven
 
-on: [push, pull_request]
+on: [push]
 
 jobs:
   build:
 
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up JDK 17
-      uses: actions/setup-java@v3
-      with:
-        distribution: 'temurin'
-        java-version: 17
-    - name: Build with Maven
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: mvn -B verify -fae --file pom.xml
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - uses: s4u/maven-settings-action@v2.8.0
```

Consider referencing a specific commit hash for third-party or even all GitHub actions, https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

> **Pin actions to a full length commit SHA**
>
> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload. When selecting a SHA, you should verify it is from the action's repository and not a repository fork.

```suggestion
      - uses: s4u/maven-settings-action@60912582505985be4cc55d2b890eb32767f8de5f #v2.8.0
```
...there is a v3.0.0 release

---

implemented with 647c5e5